include/libvfio-user.h

```diff
@@ -623,53 +623,30 @@ vfu_pci_setup_caps(vfu_ctx_t *vfu_ctx, vfu_cap_t **caps, int nr_caps);
  * Find the offset within config space of a given capability (if there are
  * multiple possible matches, use vfu_pci_find_next_capability()).
  *
- * Returns 0 if no such capability was found.
+ * Returns 0 if no such capability was found, with errno set.
  *
  * @vfu_ctx: the libvfio-user context
- * @id: capability id
+ * @extended whether capability is an extended one or not
+ * @id: capability id (PCI_CAP_ID_* or PCI_EXT_CAP_ID *)
  */
 size_t
-vfu_pci_find_capability(vfu_ctx_t *vfu_ctx, int cap_id);
+vfu_pci_find_capability(vfu_ctx_t *vfu_ctx, bool extended, int cap_id);
```

Comment:
I'm not dictating what should be done, if this is how the kernel does it then we can stick to it, I'm merely pointing out whether it's worth considering alternatives.

Reply:
No, I think your suggestion is fine. We already have this flag for adding the cap so it makes sense I think.